src/cts/src/TritonCTS.cpp

```diff
@@ -109,14 +109,72 @@ void TritonCTS::addBuilder(TreeBuilder* builder)
   builders_->push_back(builder);
 }
 
+int TritonCTS::getBufferFanoutLimit(const std::string& bufferName)
+{
+  float fanout, fanoutTemp;
+  bool hasMaxFanout, hasMaxFanoutTemp;
+  odb::dbMaster* bufferMaster = db_->findMaster(bufferName.c_str());
+  sta::Cell* bufferCell = network_->dbToSta(bufferMaster);
+  sta::Port* buffer_port = nullptr;
+  for (auto mterm : bufferMaster->getMTerms()) {
+    auto sig_type = mterm->getSigType();
+    if (sig_type == odb::dbSigType::GROUND
+        || sig_type == odb::dbSigType::POWER) {
+      continue;
+    }
+    auto io_type = mterm->getIoType();
+    if (io_type == odb::dbIoType::OUTPUT) {
+      buffer_port = network_->dbToSta(mterm);
+      break;
+    }
+  }
+  if (buffer_port == nullptr) {
+    return 0;
+  }
+  openSta_->sdc()->fanoutLimit(
```

Comment:
Just a minor suggestion.  You can also start with a large fanout limit like INF. 
Then you can start taking, sequentially one by one, min (fanout_limit, new_fanout_limit) for port, cell, liberty port, etc.
fanout_limit = INF;
if (port has fanout limit) {
    fanout_limit = MIN (fanout_limit, new fanout limit);
}
If (cell has fanout limt) {
   fanout_limit = MIN(fanout_limit, new fanout limit);
}
If (lib port has fanout limit) {
   fanout_limit = MIN(fanout_limit, new fanout limit);
}